partner_invoicing_mode/models/res_partner.py

```diff
@@ -12,10 +12,21 @@ class ResPartner(models.Model):
         default=False,
         help="Do not group sale order into one invoice.",
     )
+    next_invoice_date = fields.Date(
```

Comment:
Is this date recomputed when the invoicing mode is changed on the partner ?

Reply:
The idea here was to have a simple field that is updated after invoice generation.

Like here : https://github.com/OCA/account-invoicing/pull/1534/files#diff-d59687f72a9a4701fe1b9c5796040a8477fbb8a09c4faaa0925a1ce2293aac83R22

For the 14 days module, the next invoice date can be different on each partner (but the gap is 14 days), so I needed to store that next invoice date. As some extensions would maybe need that field too, that's why I put it on the base module (with no update).

If we change the invoicing mode, maybe indeed it should be updated to False. What do you think ?

Reply:
Only thinking about the change of behavior of the module.
Before this change we had one invoicing date (by invoicing mode), every monthly invoiced partner were invoiced on the same day.
Now we could have the monthly invoiced partner invoiced on different day.

Good, bad ? I don't know...

Should we have an option to allow for both behavior ?

Reply:
In my opinion,
- Most common case is the same day every weeek, month o whatever because in our case clients invoice monthly (for ex.) one day of the month
- Less common but Denis have this use case, every single partner has different day of invoices (which in my management opinion of a company is crazy but there are lots of optinions in the world)

So:
- We shouldn't change the behaviour, we should respect how it was working
- We should add a new behaviour for this new use case

Reply:
> Only thinking about the change of behavior of the module. Before this change we had one invoicing date (by invoicing mode), every monthly invoiced partner were invoiced on the same day (If I recall correctly). Now we could have the monthly invoiced partner invoiced on different days.
> 
> Good, bad ? I don't know...
> 
> Should we have an option to allow for both behavior ?

@TDu That behavior hasn't been changed as this function remains the same:

https://github.com/OCA/account-invoicing/blob/16.0/partner_invoicing_mode_monthly/models/sale_order.py#L69

Reply:
> * Less common but Denis have this use case, every single partner has different day of invoices (which in my management opinion of a company is crazy but there are lots of opinions in the world)

@rafaelbn Indeed, that's in customer requirements as they have different customer profiles and they want to manage it like that.

That's why I have extended the existing module in order to lower the necessary code if flow is not needed (and that's why I don't like putting everything in the same module)